frame/treasury/src/lib.rs

```diff
@@ -155,99 +206,154 @@ pub trait Trait: frame_system::Trait {
-	type Event: From<Event<Self>> + Into<<Self as frame_system::Trait>::Event>;
+	type Event: From<Event<Self, I>> + Into<<Self as frame_system::Trait>::Event>;
 
-	/// Handler for the unbalanced decrease when slashing for a rejected proposal.
-	type RingProposalRejection: OnUnbalanced<RingNegativeImbalance<Self>>;
-	/// Handler for the unbalanced decrease when slashing for a rejected proposal.
-	type KtonProposalRejection: OnUnbalancedKton<KtonNegativeImbalance<Self>>;
+	/// Handler for the unbalanced decrease when slashing for a rejected proposal or bounty.
+	type OnSlashRing: OnUnbalanced<RingNegativeImbalance<Self, I>>;
+	/// Handler for the unbalanced decrease when slashing for a rejected proposal or bounty.
+	type OnSlashKton: OnUnbalancedKton<KtonNegativeImbalance<Self, I>>;
 
 	/// Fraction of a proposal's value that should be bonded in order to place the proposal.
 	/// An accepted proposal gets these back. A rejected proposal does not.
 	type ProposalBond: Get<Permill>;
 
 	/// Minimum amount of *RING* that should be placed in a deposit for making a proposal.
-	type RingProposalBondMinimum: Get<RingBalance<Self>>;
+	type RingProposalBondMinimum: Get<RingBalance<Self, I>>;
 	/// Minimum amount of *KTON* that should be placed in a deposit for making a proposal.
-	type KtonProposalBondMinimum: Get<KtonBalance<Self>>;
+	type KtonProposalBondMinimum: Get<KtonBalance<Self, I>>;
 
 	/// Period between successive spends.
 	type SpendPeriod: Get<Self::BlockNumber>;
 
 	/// Percentage of spare funds (if any) that are burnt per spend period.
 	type Burn: Get<Permill>;
 
+	/// The amount held on deposit for placing a bounty proposal.
+	type BountyDepositBase: Get<RingBalance<Self, I>>;
+
+	/// The delay period for which a bounty beneficiary need to wait before claim the payout.
+	type BountyDepositPayoutDelay: Get<Self::BlockNumber>;
+
+	/// Bounty duration in blocks.
+	type BountyUpdatePeriod: Get<Self::BlockNumber>;
+
+	/// Percentage of the curator fee that will be reserved upfront as deposit for bounty curator.
+	type BountyCuratorDeposit: Get<Permill>;
+
+	/// Minimum value for a bounty.
+	type BountyValueMinimum: Get<RingBalance<Self, I>>;
+
+	/// Maximum acceptable reason length.
+	type MaximumReasonLength: Get<u32>;
+
 	/// Handler for the unbalanced decrease when treasury funds are burned.
-	type RingBurnDestination: OnUnbalanced<RingNegativeImbalance<Self>>;
+	type RingBurnDestination: OnUnbalanced<RingNegativeImbalance<Self, I>>;
 
 	/// Handler for the unbalanced decrease when treasury funds are burned.
-	type KtonBurnDestination: OnUnbalancedKton<KtonNegativeImbalance<Self>>;
+	type KtonBurnDestination: OnUnbalancedKton<KtonNegativeImbalance<Self, I>>;
 
 	/// Weight information for extrinsics in this pallet.
 	type WeightInfo: WeightInfo;
 }
 
 decl_storage! {
-	trait Store for Module<T: Trait> as DarwiniaTreasury {
+	trait Store for Module<T: Trait<I>, I: Instance = DefaultInstance> as DarwiniaTreasury {
```

Comment:
How different Instances allocate their storage keys prefixes?

Reply:
with suffix _InstanceXX, but I didn't see any migration in the origin code

Reply:
DefaultInstance is fine